tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+	for _, group := range reportData.Results {
+		for _, check := range group.Checks {
+			assert.NotEqualf(c.T(), "fail", check.State, "check failed: id: %s state: %s description: %s ", check.Id, check.State, check.Description)
+		}
+	}
+
+	require.Equalf(c.T(), reportData.Fail, 0, "cluster scan failed")
+}
+
+func (c *CisBenchmarkTestSuite) TestUpgradeCisbenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	// Change cisbenchmark install option version to previous version of the latest version
+	versionsList, err := client.Catalog.GetListChartVersions(charts.RancherCisBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+
+	if len(versionsList) < 2 {
+		c.T().Skip("Skipping the upgrade case, only one version of cis-benchmark is available")
+	}
+	assert.GreaterOrEqualf(c.T(), len(versionsList), 2, "There should be at least 2 versions of the cis-benchmark chart")
+	versionLatest := versionsList[0]
+	c.T().Log(versionLatest)
+
+	versionBeforeLatest := versionsList[1]
+	c.T().Log(versionBeforeLatest)
+	c.chartInstallOptions.Version = versionBeforeLatest
+
+	c.T().Log("Checking if the cis-benchmark chart is installed with one of the previous versions")
+	initialCisbenchmarkChart, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	if initialCisbenchmarkChart.IsAlreadyInstalled && initialCisbenchmarkChart.ChartDetails.Spec.Chart.Metadata.Version == versionLatest {
+		c.T().Skip("Skipping the upgrade case, cis-benchmark chart is already installed with the latest version")
+	}
+
+	if !initialCisbenchmarkChart.IsAlreadyInstalled {
+		c.T().Log("Installing cis-benchmark chart with the version before the latest version")
+		err = charts.InstallRancherCisBenchmarkChart(client, c.chartInstallOptions)
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cis-benchmark chart deployments to have expected number of available replicas")
+		err = charts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cisbenchmark chart DaemonSets to have expected number of available nodes")
+		err = charts.WatchAndWaitDaemonSets(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+	}
+
+	cisbenchmarkChartPreUpgrade, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	// Validate current version of rancher-cis-benchmark is one of the versions before latest
+	chartVersionPreUpgrade := cisbenchmarkChartPreUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Contains(c.T(), versionsList[1:], chartVersionPreUpgrade)
```

Comment:
shouldn't this involve taking a scan so that the chart has been used in the test instead of just installed?

Reply:
Thanks for pointing it out. I updated the code to run the scan after installation and also post upgrade.